### PR TITLE
[86bxtqayu][utils] fixed condition to check on which element to move focus to in a locked container

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
I've changed a condition to move focus to the last element in locked container - for now we check that current active node is follow previous active node and current active node is a descendant of the previous node.

## How has this been tested?
I've tested it manually

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/utils/src/use/useFocusLock.ts [hunk 1]
```diff
@@ -219,6 +206,30 @@ const useFocusLockHook = (
     };
   }, [disabled, autoFocus, returnFocusTo, returnFocus]);
 
+  React.useEffect(() => {
```
Comment (msereniti) [inline semcore/utils/src/use/useFocusLock.ts:209]: Why did you move the effect?

Comment (ilyabrower) [reply semcore/utils/src/use/useFocusLock.ts:209]: because I need to compare values in focusMastersStack in `returnFocus` before they removed from stack

**Review by msereniti — APPROVED** [review]

**Review by msereniti — APPROVED** [review]
